assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py

```diff
@@ -383,8 +383,10 @@ def _get_local_file_list(start_datetime: datetime, end_datetime: datetime, input
     if end_datetime.minute == 0 and end_datetime.second == 0:
         # if end_datetime is a whole hour, the last hour folder is not needed
         end_datetime -= timedelta(seconds=1)
-    cur_datetime = start_datetime
-    while cur_datetime <= end_datetime:
+
```

Comment:
P2: actually I revisit the get_file_list(), seems it also works for local file, maybe you can comment below three lines and have a test:
```py
    if store_url.is_local_path():
        # for local testing
        return _get_local_file_list(start_datetime, end_datetime, store_url._base_url)
```

Reply:
Let me test this, it would be good to have the same logic

Reply:
Seems that it works locally. Let me check if it works for github.

Reply:
I supposed you also verified in github?

Reply:
Right. The tests ran fine on CI